### PR TITLE
Add `list worker types` endpoint to worker manager

### PR DESCRIPTION
**Pull request by owlishDeveloper**
Bugzilla Bug: [1546789](https://bugzilla.mozilla.org/show_bug.cgi?id=1546789)

### CODE REVIEW

File: services/worker-manager/src/api.js [hunk 1]
```diff
@@ -165,3 +165,32 @@ builder.declare({
...
+  const limit = parseInt(req.query.limit || 100, 10);
+  const scanOptions = {
+    continuation: continuationToken,
+    limit,
+  };
+
+  const data = await this.WorkerType.scan({}, scanOptions);
+
+  if (data.continuation) {
+    data.continuationToken = data.continuation;
```
Comment (owlishDeveloper) [inline services/worker-manager/src/api.js]: not sure about this 🤔 your thoughts @imbstack ?

Comment (imbstack) [reply services/worker-manager/src/api.js]: Fine with me!

File: services/worker-manager/src/api.js [hunk 2]
```diff
@@ -165,3 +165,32 @@ builder.declare({
...
+  route: '/workertypes',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name: 'listWorkerTypes',
+  title: 'List All WorkerTypes',
+  stability: APIBuilder.stability.experimental,
+  description: [
+    'TODO',
```
Comment (imbstack) [inline services/worker-manager/src/api.js]: Let's add a description and get this landed!

Only other thing is an output schema. It should just be an array of the workertype-full objects. You can use `$ref` to set that up.

File: services/worker-manager/src/api.js [hunk 3]
```diff
@@ -165,3 +165,32 @@ builder.declare({
...
+builder.declare({
+  method: 'get',
+  route: '/workertypes',
+  query: {
+    continuationToken: /./,
+    limit: /^[0-9]+$/,
+  },
+  name: 'listWorkerTypes',
+  title: 'List All WorkerTypes',
+  stability: APIBuilder.stability.experimental,
```
Comment (imbstack) [inline services/worker-manager/src/api.js]: Just need to add the new schema as `output: ...`.

File: services/worker-manager/schemas/v1/workertype-list [hunk 4]
```diff
@@ -0,0 +1,27 @@
+$schema: "/schemas/common/metaschema.json#"
+title: WorkerType Full
```
Comment (imbstack) [inline services/worker-manager/schemas/v1/workertype-list]: WorkerType List

File: services/worker-manager/schemas/v1/workertype-list [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+description: A complete workerType definition
```
Comment (imbstack) [inline services/worker-manager/schemas/v1/workertype-list]: A list of workerType definitions

File: services/worker-manager/schemas/v1/workertype-list (hunk 4 above)
Comment (imbstack) [reply services/worker-manager/schemas/v1/workertype-list]: We need to add the `.yml` at the end of this filename.

File: services/worker-manager/schemas/v1/workertype-list [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+type: object
+properties:
+  data:
```
Comment (imbstack) [inline services/worker-manager/schemas/v1/workertype-list]: Let's have this key be `workertypes` to keep the same format as the list of clients from auth has for instance.

File: services/worker-manager/schemas/v1/workertype-list [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    description: |
+      Opaque `continuationToken` to be given as query-string option to get the
+      next set of worker-types in the worker-manager.
+      This property is only present if another request is necessary to fetch all
+      results. In practice the next request with a `continuationToken` may not
+      return additional results, but it can. Thus, you can only be sure to have
+      all the results if you've called `listWorkerTypes` with `continuationToken`
+      until you get a result without a `continuationToken`.
+
+additionalProperties: true
```
Comment (imbstack) [inline services/worker-manager/schemas/v1/workertype-list]: This should be false, right?

**Comment by imbstack** [comment]
It occurs to me that this was probably still WIP when I did my followup review and you hadn't requested me to look at it yet. Sorry to jump the gun.

File: services/worker-manager/schemas/v1/workertype-list (hunk 7 above)
Comment (owlishDeveloper) [reply services/worker-manager/schemas/v1/workertype-list]: https://github.com/taskcluster/taskcluster/pull/632#discussion_r279103307 
I understand there also will be `continuation` prop

**Comment by owlishDeveloper** [comment]
This PR was never a WIP

**Pull request merged**